ffi/include/wirefilter.h

```diff
@@ -118,6 +118,14 @@ bool wirefilter_filter_uses(
     wirefilter_externally_allocated_str_t field_name
 );
 
+uint64_t wirefilter_get_filter_hash(const wirefilter_filter_ast_t *ast);
+
+wirefilter_rust_allocated_str_t wirefilter_serialize_filter_to_json(
+    const wirefilter_filter_ast_t *ast
+);
+
+void wirefilter_free_string(wirefilter_rust_allocated_str_t str);
```

Comment:
Now that we have this, probably worth replacing `wirefilter_free_parsing_result` with just `wirefilter_free_ast`? (for the success case only)

@inikulin what do you think?

Reply:
You mean like having to call explicitly `wirefilter_free_ast(result.ok.ast)` if `result.success` is `true` and `wirefilter_free_string(result.err.msg)` if not ?

To be honest, i don't really like the whole parsing result structure, but i am not sure you can safely get rid of it.
I thought of something like
```Rust
wirefilter_filter_ast_t *wirefilter_parse_filter(
    const wirefilter_scheme_t *scheme,
    wirefilter_externally_allocated_str_t input,
    wirefilter_rust_allocated_str_t *error,
);
```
But that would imply to return NULL pointer in case of errors

Reply:
> You mean like having to call explicitly wirefilter_free_ast(result.ok.ast) if result.success is true and wirefilter_free_string(result.err.msg) if not ?

Yep, that. Especially because now we have `compile_...` that takes ownership of AST, so sometimes you want to free *only* in case of an error.

Reply:
If we decide to do a braking change, I on the same boat as @marmeladema and would rather get rid of the parsing result structure entirely using `get_last_error` approach.

Reply:
We talked about that, and we can't really do that because errors returned from different APIs can have different types and details.